stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java

```diff
@@ -94,14 +112,96 @@ public static MimeMessage generateMessage(Session session, String username, MapV
+                } else {
+                    multipart.addBodyPart(buildJavaMailBodyPart(mimeEntity, contentType));
+                }
+            }
+        }
+        emailMessage.setContent(multipart);
+    }
+
+    private static MimeBodyPart populateMultipart(ObjectValue mimeEntity) throws IOException, MessagingException {
+        Multipart multipart = new MimeMultipart();
+        ArrayValue multipartMimeEntityArrayValue = EntityBodyHandler.getBodyPartArray(mimeEntity);
+        int entityCount = multipartMimeEntityArrayValue.size();
+        for (int i = 0; i < entityCount; i++) {
+            ObjectValue childMimeEntity = (ObjectValue) multipartMimeEntityArrayValue.get(i);
+            String childContentType = getContentTypeWithParameters(childMimeEntity);
+            if (childContentType.startsWith(MimeConstants.MULTIPART_AS_PRIMARY_TYPE)) {
+                multipart.addBodyPart(populateMultipart(childMimeEntity));
+            } else {
+                multipart.addBodyPart(buildJavaMailBodyPart(childMimeEntity, childContentType));
+            }
+        }
+        MimeBodyPart returnMimeBodyPart = new MimeBodyPart();
+        returnMimeBodyPart.setContent(multipart);
+        return returnMimeBodyPart;
+    }
+
+    private static MimeBodyPart buildJavaMailBodyPart(ObjectValue mimeEntity, String contentType)
+            throws MessagingException, IOException {
+        MimeBodyPart attachmentBodyPart = new MimeBodyPart();
+        Channel channel = EntityBodyHandler.getByteChannel(mimeEntity);
+        if (channel != null) {
+            InputStream inputStream = channel.getInputStream();
+            ByteArrayDataSource ds = new ByteArrayDataSource(inputStream, contentType);
+            attachmentBodyPart.setDataHandler(new DataHandler(ds));
+        } else {
+            if (CommonUtil.isTextBased(contentType)) {
+                attachmentBodyPart.setText((String) MimeDataSourceBuilder.getText(mimeEntity));
+            } else {
+                ArrayValue binaryContent = (ArrayValue) MimeDataSourceBuilder.getByteArray(mimeEntity);
+                attachmentBodyPart.setContent(binaryContent.getBytes(), MimeConstants.OCTET_STREAM);
+            }
+        }
+        addHeadersToJavaMailBodyPart(mimeEntity, attachmentBodyPart);
+        return attachmentBodyPart;
+    }
+
+    private static void addHeadersToJavaMailBodyPart(ObjectValue mimeEntity, MimeBodyPart attachmentBodyPart)
+            throws MessagingException {
+        ArrayValue headerNamesArrayValue = EntityHeaders.getHeaderNames(mimeEntity, MimeConstants.LEADING_HEADER);
+        HandleValue[] handleValues = (HandleValue[]) headerNamesArrayValue.getValues();
+        String[] headerNames = new String[handleValues.length];
+        for (int j = 0; j < handleValues.length; j++) {
+            headerNames[j] = handleValues[j].stringValue();
+        }
+        if (headerNames.length > 0) {
+            for (String headerName : headerNames) {
```

Comment:
Shall we replace this with a `for each` loop?

Reply:
Can you suggest the code?